src/basic/strv.c

```diff
@@ -353,6 +353,58 @@ int strv_split_extract(char ***t, const char *s, const char *separators, Extract
         return (int) n;
 }
 
+int strv_split_extract_pairs_colon(char ***t, const char *s) {
+        _cleanup_strv_free_ char **l = NULL;
+        size_t n = 0, allocated = 0;
+        int r;
+
+        assert(t);
+        assert(s);
+
+        for (;;) {
+                _cleanup_free_ char *first = NULL, *second = NULL, *tuple = NULL, *second_or_empty = NULL;
+                const char *p = NULL;
+
+                r = extract_first_word(&s, &tuple, WHITESPACE, EXTRACT_CUNESCAPE|EXTRACT_CUNESCAPE_RELAX|EXTRACT_UNQUOTE|EXTRACT_DONT_COALESCE_SEPARATORS);
```

Comment:
`EXTRACT_CUNESCAPE_RELAX` seems iffy — we mostly use that for compat. Here we should just require people to use correct syntax from the start.
`EXTRACT_DONT_COALESCE_SEPARATORS` also. Additional whitespace shouldn't matter.

Reply:
Without _RELAX, the test input "seven:eight\\:nine" causes an EINVAL to be returned from here: https://github.com/systemd/systemd/blob/master/src/basic/extract-word.c#L99

Reply:
right, it's because ```\:``` is not a normal C escape sequence, so it doesn't recognize it

Reply:
cherry-picked your commit from https://github.com/systemd/systemd/pull/16655 - thanks!